tests/specs/mcd/word-pack.k

```diff
@@ -46,26 +46,113 @@ endmodule
 module WORD-PACK-HASKELL [kore]
     imports WORD-PACK-COMMON
 
-    rule    ADDR_UINT48_UINT48 ==Int #WordPackAddrUInt48UInt48(ADDR, UINT48_1, UINT48_2)
-         => ADDR     ==Int maxUInt160 &Int  ADDR_UINT48_UINT48
-    andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
-    andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
-    andBool #rangeUInt(256, ADDR_UINT48_UINT48)
-      [simplification]
-
-    rule    UINT48_UINT48 ==Int #WordPackUInt48UInt48(UINT48_1, UINT48_2)
-         => UINT48_1 ==Int maxUInt48 &Int  UINT48_UINT48
-    andBool UINT48_2 ==Int maxUInt48 &Int (UINT48_UINT48 /Int pow48)
-    andBool #rangeUInt(96, UINT48_UINT48)
-      [simplification]
+    rule       ADDR_UINT48_UINT48:Int ==Int #WordPackAddrUInt48UInt48(ADDR:Int, UINT48_1:Int, UINT48_2:Int) 
+            => ADDR     ==Int maxUInt160 &Int  ADDR_UINT48_UINT48              
+       andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
+       andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
+       andBool #rangeUInt(256, ADDR_UINT48_UINT48)
+       andBool #rangeUInt(160, ADDR)
+       andBool #rangeUInt(48, UINT48_1)
+       andBool #rangeUInt(48, UINT48_2)
+       [simplification]
+
+    rule       #WordPackAddrUInt48UInt48(ADDR:Int, UINT48_1:Int, UINT48_2:Int) ==Int ADDR_UINT48_UINT48:Int
+            => ADDR     ==Int maxUInt160 &Int  ADDR_UINT48_UINT48              
+       andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
+       andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
+       andBool #rangeUInt(256, ADDR_UINT48_UINT48)
+       andBool #rangeUInt(160, ADDR)
+       andBool #rangeUInt(48, UINT48_1)
+       andBool #rangeUInt(48, UINT48_2)
+       [simplification]
+
+    rule    { ADDR_UINT48_UINT48:Int #Equals #WordPackAddrUInt48UInt48(ADDR:Int, UINT48_1:Int, UINT48_2:Int) }
+         => { ADDR     #Equals maxUInt160 &Int  ADDR_UINT48_UINT48              }
+       #And { UINT48_1 #Equals maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160) }
+       #And { UINT48_2 #Equals maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208) }
+       #And { #rangeUInt(256, ADDR_UINT48_UINT48) }
+       #And { #rangeUInt(160, ADDR) }
+       #And { #rangeUInt(48, UINT48_1) }
+       #And { #rangeUInt(48, UINT48_2) }
```

Comment:
What are these? `{ ... }` doesn't mean anything I don't think. Perhaps `{ true #Equals ... }` instead?

Reply:
Also, are we sure that these lemmas over `{ ... #Equals ... }` are necessary?

Reply:
All of these have been adjusted to be of the form `{ true #Equals X }`, where `X` lives in Bool, meaning no `#And`, `#Or`, and the like...